File: src/main/java/com/company/erp/features/users/dto/UpdateUserRequest.java
package com.company.erp.features.users.dto;

import jakarta.validation.constraints.Size;

import java.util.Set;

public record UpdateUserRequest(
        @Size(max = 255) String fullName,
        @Size(max = 50)  String phone,
        @Size(max = 1024) String avatarUrl,
        Boolean enabled,
        /** Replaces the user's role set. {@code null} means "leave roles untouched." */
        Set<String> roles
) {
}

File: app/Features/Users/Requests/UpdateUserRequest.php
<?php

namespace App\Features\Users\Requests;

use App\Support\Http\ApiFormRequest;

class UpdateUserRequest extends ApiFormRequest
{
    public function rules(): array
    {
        return [
            'fullName' => ['sometimes', 'nullable', 'string', 'max:255'],
            'phone' => ['sometimes', 'nullable', 'string', 'max:50'],
            'avatarUrl' => ['sometimes', 'nullable', 'string', 'max:1024'],
            'enabled' => ['sometimes', 'nullable', 'boolean'],
            'roles' => ['sometimes', 'nullable', 'array'],
            'roles.*' => ['string'],
        ];
    }
}
